src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -235,7 +235,19 @@ void InferRequest::set_input(const std::string& name, const Blob::Ptr& data) {
     } else {
         if (data->buffer() == nullptr)
             IE_THROW(NotAllocated) << str_input_not_allocated << " Input name: \'" << name << "\'";
-        _inputs[name] = data;
+        if (_inputs.find(name) != _inputs.end() && _inputs[name]->byteSize() == data->byteSize()) {
+            auto src_lock = data->cbuffer();
+            auto src_ptr = src_lock.as<uint8_t*>();
+            auto tgt_lock = _inputs[name]->cbuffer();
+            auto tgt_ptr = tgt_lock.as<uint8_t*>();
```

Comment:
[random spot] Need a test

Reply:
These changes are removed.

Reply:
still need a test

Reply:
I added a test to set tensor of f64 scalar type as input, which fails without this patch. Thank you.